server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MultipleDatabaseChangeRuntimeFlowableTask.java

```diff
@@ -325,4 +380,29 @@ public List<Class<? extends ExecutionListener>> getExecutionListenerClasses() {
         return null;
     }
 
+    public List<Long> listNeedCancelIds(List<Long> instanceIds) {
+        // Clear the level 1 cache of JPA
+        entityManager.clear();
+        List<FlowInstanceEntity> flowInstanceEntityList = flowInstanceService.listByIds(instanceIds);
```

Comment:
may simplify use 'select id from flow_instance where id in () and status in ()' .

Reply:
ok, it's better to do that when querying the database